src/test/java/cloud/fogbow/ras/core/plugins/interoperability/cloudstack/compute/v4_9/CloudStackComputePluginTest.java

```diff
@@ -1,894 +1,1426 @@
-    public static final String USER_DATA_KEY = "userdata";
-    public static final String CLOUDSTACK_URL = "cloudstack_api_url";
-    public static final String CLOUD_NAME = "cloudstack";
-
-    private String fakeZoneId;
+@PrepareForTest({SharedOrderHolders.class, CloudStackUrlUtil.class, GetVolumeResponse.class,
+        DefaultLaunchCommandGenerator.class, PropertiesUtil.class, GetVirtualMachineResponse.class,
+        DatabaseManager.class, DeployVirtualMachineResponse.class})
+public class CloudStackComputePluginTest extends BaseUnitTests {
+
+    private static final CloudStackUser CLOUD_STACK_USER =
+            new CloudStackUser("", "", "", "", new HashMap<>());
+    private static final String BAD_REQUEST_MSG = "Bad Request";
+    private final int AMOUNT_EXTRA_SERVICE_OFFERING = 4;
+    private static final String FAKE_NETWORK_ID = "fake-network-id";
+    private static final String CLOUDSTACK_CLOUD_NAME = "cloudstack";
 
     private CloudStackComputePlugin plugin;
     private CloudStackHttpClient client;
     private LaunchCommandGenerator launchCommandGeneratorMock;
     private Properties properties;
     private String defaultNetworkId;
-    private SharedOrderHolders sharedOrderHolders;
+
+    @Rule
+    private ExpectedException expectedException = ExpectedException.none();
 
     @Before
-    public void setUp() {
-        String cloudStackConfFilePath = HomeDir.getPath() + SystemConstants.CLOUDS_CONFIGURATION_DIRECTORY_NAME +
-                File.separator + CLOUD_NAME + File.separator + SystemConstants.CLOUD_SPECIFICITY_CONF_FILE_NAME;
+    public void setUp() throws UnexpectedException {
+        String cloudStackConfFilePath = HomeDir.getPath() +
+                SystemConstants.CLOUDS_CONFIGURATION_DIRECTORY_NAME + File.separator
+                + CLOUDSTACK_CLOUD_NAME + File.separator
+                + SystemConstants.CLOUD_SPECIFICITY_CONF_FILE_NAME;
         this.properties = PropertiesUtil.readProperties(cloudStackConfFilePath);
-        this.defaultNetworkId = this.properties.getProperty(CloudStackPublicIpPlugin.DEFAULT_NETWORK_ID_KEY);
+        this.defaultNetworkId = this.properties.getProperty(
+                CloudStackPublicIpPlugin.DEFAULT_NETWORK_ID_KEY);
         this.launchCommandGeneratorMock = Mockito.mock(LaunchCommandGenerator.class);
         this.client = Mockito.mock(CloudStackHttpClient.class);
-        this.plugin = new CloudStackComputePlugin(cloudStackConfFilePath);
+        this.plugin = Mockito.spy(new CloudStackComputePlugin(cloudStackConfFilePath));
         this.plugin.setClient(this.client);
         this.plugin.setLaunchCommandGenerator(this.launchCommandGeneratorMock);
-        this.fakeZoneId = this.properties.getProperty(CloudStackComputePlugin.ZONE_ID_KEY);
+        this.testUtils.mockReadOrdersFromDataBase();
+    }
+
+    // test case: successffuly case
+    @Test
+    public void testDoGetInstance() throws FogbowException {
```

Comment:
you have to check if requestGetVirtualMachine() and buildComputeInstance() have been called here. That's the only "contract" a doGetInstance() call must follow.

Reply:
This was indirectly checked when It was done the mock.